cylc/flow/task_pool.py

```diff
@@ -705,14 +714,13 @@ def _get_task_by_id(self, id_):
             except KeyError:
                 pass
 
-    def queue_tasks(self, itasks):
-        """Queue tasks that are ready to run."""
-        for itask in itasks:
-            itask.state.reset(is_queued=True)
-            # TODO Reset manual trigger flag. One manual trigger queues an
-            # unqueued task, another one triggers a queued task.
-            self.data_store_mgr.delta_task_queued(itask)
-        self.task_queue_mgr.push_tasks(itasks)
+    def queue_task(self, itask: TaskProxy) -> None:
+        """Queue a task that is ready to run."""
+        itask.state.reset(is_queued=True)
+        # TODO Reset manual trigger flag. One manual trigger queues an
+        # unqueued task, another one triggers a queued task.
```

Comment:
(FYI this TODO was addressed in #4103 but I evidently neglected to remove the comment ... now done, if you update your branch it will disappear).